### PR TITLE
Upgrade to new jcasc test harness

### PR DESCRIPTION
**Pull request by timja**
JCasc had to create a new dependency that wasn't a `tests` classifier because maven doesn't allow you to bring dependencies through classifiers.

Upgrading here so that PCT works again,

Tracking:
https://github.com/jenkinsci/bom/pull/164

### CODE REVIEW

File: src/test/java/com/microsoftopentechnologies/windowsazurestorage/helper/CredentialRenameTest.java [hunk 1]
```diff
@@ -5,80 +5,29 @@
...
-import java.io.File;
-import java.util.Collections;
-import java.util.List;
+import org.jvnet.hudson.test.recipes.LocalData;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 
 public class CredentialRenameTest {
-    public static File input = new File("src/test/resources/com/microsoftopentechnologies/helper/correctFormatOldConfig.xml");
```
Comment (timja) [inline src/test/java/com/microsoftopentechnologies/windowsazurestorage/helper/CredentialRenameTest.java]: this was an unused field

File: src/test/java/com/microsoftopentechnologies/windowsazurestorage/helper/CredentialRenameTest.java [hunk 2]
```diff
@@ -5,80 +5,29 @@
...
-                    "            <storageAccountKey>key</storageAccountKey>\n" +
-                    "            <blobEndpointURL>https://blob.core.windows.net/</blobEndpointURL>\n" +
-                    "          </storageData>\n" +
-                    "        </com.microsoftopentechnologies.windowsazurestorage.helper.AzureCredentials>\n" +
-                    "      </java.util.concurrent.CopyOnWriteArrayList>\n" +
-                    "    </entry>\n" +
-                    "  </domainCredentialsMap>\n" +
-                    "</com.cloudbees.plugins.credentials.SystemCredentialsProvider>";
-
-    private static final String wrongConfigContent = "<?xml version='1.0' encoding='UTF-8'?>\n" +
```
Comment (timja) [inline src/test/java/com/microsoftopentechnologies/windowsazurestorage/helper/CredentialRenameTest.java]: this was an unused field

File: src/test/java/com/microsoftopentechnologies/windowsazurestorage/helper/CredentialRenameTest.java [hunk 3]
```diff
@@ -5,80 +5,29 @@
...
-import java.util.List;
+import org.jvnet.hudson.test.recipes.LocalData;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 
 public class CredentialRenameTest {
-    public static File input = new File("src/test/resources/com/microsoftopentechnologies/helper/correctFormatOldConfig.xml");
-    private Jenkins jenkinsInstance;
-    private static final String correctConfigContent =
```
Comment (timja) [inline src/test/java/com/microsoftopentechnologies/windowsazurestorage/helper/CredentialRenameTest.java]: on new credentials it seems like the credential.xml was cached in memory and credentials weren't being loaded by just overwriting it on the file system

File: src/test/java/com/microsoftopentechnologies/windowsazurestorage/helper/CredentialRenameTest.java [hunk 4]
```diff
@@ -5,80 +5,29 @@
...
+    @LocalData
     public void testRenameStorageConfig() throws Exception {
         String storageAccount = "name";
         String storageAccountKey = "key";
         String storageBlobURL = "https://blob.core.windows.net/";
-        File configFile = testFolder.newFile("credentials.xml");
-        FileUtils.writeStringToFile(configFile, correctConfigContent);
-        FileUtils.copyFileToDirectory(configFile, jenkinsInstance.root);
 
-        CredentialRename.renameStorageConfig();
```
Comment (timja) [inline src/test/java/com/microsoftopentechnologies/windowsazurestorage/helper/CredentialRenameTest.java]: this is called on startup, no need to call it again

**Comment by timja** [comment]
Would be great to get a release with this in, so it can be used in PCT @gavinfish  😄

File: pom.xml [hunk 5]
```diff
@@ -93,7 +93,7 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>credentials</artifactId>
-            <version>2.1.16</version>
+            <version>2.2.0</version>
```
Comment (timja) [inline pom.xml]: The credentials configurator moved to the credentials plugin from the support plugin in this version

**Comment by gavinfish** [comment]
> Would be great to get a release with this in, so it can be used in PCT @gavinfish 😄

Sure, I will draft a release.

**Pull request merged**